doc/kuksa_requirementes.md

```diff
@@ -0,0 +1,805 @@
+  For VISSv2 specifically, see ([read](https://w3c.github.io/automotive/spec/VISSv2_Core.html#read), [subscribe](https://w3c.github.io/automotive/spec/VISSv2_Core.html#subscribe)), [update](https://w3c.github.io/automotive/spec/VISSv2_Core.html#update))).
+
+VSS and VISS does _not_ specify how signals are provided _into_ the VSS server / model.
+
+These two aspects of interacting with the signals can thus be divided into:
+* **Provider**<br>
+  _A provider is providing a signal (sensor, attribute or actuator) to the VSS tree._<br>
+  _A provider can also use signals as any other user._
+
+* **User**<br>
+  _A user (or client) is using the signals in the VSS tree (without providing any of them itself)._
+
+where the VSS and VISS* specifications only specify the **User** part.
+
+When designing the databroker API, the method kuksa.val uses for providing signals was investigated. In short, kuksa-val-server takes the approach of trying to shoehorn the need of signal providers into the existing VISSv2 protocol. This is problematic for several reasons:
+
+* By reusing, **but changing the meaning** of the terminology used in VSS, there is a constant dissonance between the terms used. This is a recipe for confusion.
+* By deviating from the standard in _this particular way_, a standards compliant VISSv2 client cannot use it in any meaningful way.
+* It makes it harder to actually provide a standards compliant VISSv2 in the future .
+* By using the same methods for both signal providers _and_ users of signals it's harder (or impossible) to customize them for their different requirements.
+
+With this in mind, databroker chose to make a clear distinction between signal providers and signal users. It doesn't use this terminology though. It does this by splitting the interface into two separate services, which are customized for their different requirements / use cases. It doesn't need to be implemented in this way in order to achieve the same design goal, though.
+
+## Enable "easy to use" user facing API
+
+This is meant to illustrate what type of user APIs that can be created depending on what the provider API looks like (assuming we have one).
+
+Use case:
+
+The user wants to lock a door and know when it's done / whether it worked.
+
+**The kuksa.val.v1 or "key-value store" way.**
+
+Something conceptually like this:
+1. User calls `set(Vehicle.Door.Locked, field=TARGET_VALUE, true)`
+2. User calls `subscribe(Vehicle.Door.Locked, field=VALUE)`
+3. Provider (subscribing to TARGET_VALUE) receives the request and starts actuating, providing VALUE when it changes.
+4. User is notified when VALUE turns to true, and concludes that the door has now been locked.
+
+But what happens if the door fails to lock, e.g. the door is not close or the actuator is broken?
+* What should the user subscribe to for this information?
+* And how long should it wait before concluding that it failed?
```

Comment:
The VSS defines an explicit defect sensor signal for some actuators (like the front beams), i.e. `Vehicle.Body.Lights.Beam.Low.IsDefect` of boolean type. But that's not generally the case. 

We could discuss introducing a general defect failure state in addition to the four failure states proposed above. Setting a target value to an actuator in defect state could then be rejected by the broker/provider with a proper error.

But that wouldn't help in situations where the requested target value cannot be "achieved" for other reasons, e.g. seat or window movement is stopped due to a blockage; or a remotely requested seat movement is rejected because the vehicle is currently being driven. Currently, one could just recognize that the seat/window movement is stopped (before reaching the target) or not happening at all via the updates of the current value. (For other actuators, VSS explicitly distinguishes between current and target values). 

I guess we should not try to fix all the shortcomings of the VSS spec ...

BTW: Door's lock and open states are not necessarily related: You could lock an open door and if you close it, you can't reopen it (because of the lock). The actual behaviour will be vehicle specific.